daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -990,30 +985,28 @@ object Main extends Logging {
                   // not even 1 more bit is available.
                   // do not try to keep parsing, nothing left to parse
                   keepParsing = false
-                  error = false
                 } else {
                   // There is more data available.
                   if (parseOpts.stream.toOption.get) {
                     // Streaming mode
                     if (lastParseBitPosition == loc.bitPos0b) {
                       // this parse consumed no data, that means this would get
                       // stuck in an infinite loop if we kept trying to stream,
-                      // so we need quit
+                      // so we need to quit
                       val remainingBits =
                         if (loc.bitLimit0b.isDefined) {
                           (loc.bitLimit0b.get - loc.bitPos0b).toString
                         } else {
                           "at least " + (inStream.inputSource.bytesAvailable * 8)
                         }
-                      log(LogLevel.Warning, "Left over data after consuming 0 bits while streaming. Stopped after consuming %s bit(s) with %s bit(s) remaining.", loc.bitPos0b, remainingBits)
+                      log(LogLevel.Error, "Left over data after consuming 0 bits while streaming. Stopped after consuming %s bit(s) with %s bit(s) remaining.", loc.bitPos0b, remainingBits)
                       keepParsing = false
-                      error = true
+                      exitCode = ExitCode.NoForwardProgress
```

Comment:
I wonder if this should just be `ExitCode.LeftOverData` and we can remove `NoForwardProgress`.